src/segy/transforms.py

```diff
@@ -0,0 +1,149 @@
+"""Transforms to apply to arrays and structured arrays."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from numpy.typing import NDArray
+
+    from segy.schema import Endianness
+
+
+class TransformStrategy:
+    """Base class for header transformation strategies."""
+
+    def transform(self, data: NDArray[Any]) -> NDArray[Any]:
+        """Forward transformation implementation."""
+        raise NotImplementedError
+
+    def inverse_transform(self, data: NDArray[Any]) -> NDArray[Any]:
+        """Inverse transformation implementation."""
+        raise NotImplementedError
+
+
+class ScaleStrategy(TransformStrategy):
+    """Transform to scale a numpy array.
+
+    Args:
+        scale_factor: Scalar to use in transformations.
+    """
+
+    def __init__(self, scale_factor: int | float):
+        self.scale_factor = scale_factor
+
+    def transform(self, data: NDArray[Any]) -> NDArray[Any]:
+        """Forward transformation implementation."""
+        return data * self.scale_factor
+
+    def inverse_transform(self, data: NDArray[Any]) -> NDArray[Any]:
+        """Inverse transformation implementation."""
+        return data / self.scale_factor
+
+
+class ScaleFieldStrategy(TransformStrategy):
+    """Transform to scale user defined keys in a structured dtype.
+
+    Args:
+        scale_factor: Scalar to use in transformations.
+        keys: List of fields to apply the transformation on.
+    """
+
+    def __init__(self, scale_factor: int | float, keys: list[str]):
+        self.scale_factor = scale_factor
+        self.keys = keys
+
+    def transform(self, data: NDArray[Any]) -> NDArray[Any]:
+        """Forward transformation implementation."""
+        if data.dtype.names is None:
+            msg = f"{self.__class__.__name__} can only work on structured arrays."
+            raise ValueError(msg)
+
+        for key in self.keys:
+            if key in data.dtype.names:
+                data[key] = [vals * self.scale_factor for vals in data[key]]
```

Comment:
Can we do the multiplication on the array directly rather than on each value? Or is it not guaranteed to be each field is an array.
Eg.  `data[key] = data[key] * scale_factor`

Reply:
actually we can, not sure why I did this lol. lemme fix it and run the tests.

Reply:
the current implementation would be stupid slow, glad you caught it.